src/SAML2/XML/saml/BaseID.php

```diff
@@ -8,91 +8,52 @@
 use SimpleSAML\Assert\Assert;
 use SimpleSAML\SAML2\Compat\ContainerSingleton;
 use SimpleSAML\SAML2\Constants as C;
+use SimpleSAML\SAML2\Utils;
+use SimpleSAML\SAML2\XML\ExtensionPointInterface;
+use SimpleSAML\SAML2\XML\ExtensionPointTrait;
 use SimpleSAML\SAML2\XML\IDNameQualifiersTrait;
 use SimpleSAML\XML\Exception\InvalidDOMElementException;
-use SimpleSAML\XML\XMLStringElementTrait;
+use SimpleSAML\XML\Exception\SchemaViolationException;
 use SimpleSAML\XMLSecurity\Backend\EncryptionBackend;
 use SimpleSAML\XMLSecurity\XML\EncryptableElementInterface;
 use SimpleSAML\XMLSecurity\XML\EncryptableElementTrait;
 
-use function trim;
+use function count;
+use function explode;
 
 /**
  * SAML BaseID data type.
  *
  * @package simplesamlphp/saml2
  */
-class BaseID extends AbstractSamlElement implements BaseIdentifierInterface, EncryptableElementInterface
+class BaseID extends AbstractSamlElement implements
```

Comment:
Shall we make it abstract?

Reply:
Depends; if we want to instantiate unknown BaseID from the :fromXML(), it cannot be abstract.
If we go with a separate UnknownBaseID class, we can do it..  The constructor is already protected, so it cannot be instantiated immediately